src/lib/y2network/autoinst/interfaces_reader.rb

```diff
@@ -0,0 +1,150 @@
+
+          case config
+          when ConnectionConfig::Vlan
+            load_vlan(config, interface_section)
+          when ConnectionConfig::Bridge
+            load_bridge(config, interface_section)
+          when ConnectionConfig::Bonding
+            load_bonding(config, interface_section)
+          when ConnectionConfig::Wireless
+            load_wireless(config, interface_section)
+          end
+
+          log.info "Resulting config: #{config.inspect}"
+          config
+        end
+
+        ConnectionConfigsCollection.new(configs)
+      end
+
+    private
+
+      def create_config(interface_section)
+        # TODO: autoyast backend for type detector?
+        # TODO: TUN/TAP interface missing for autoyast?
+        return ConnectionConfig::Bonding.new if interface_section.bonding_slave0 && !interface_section.bonding_slave0.empty?
+        return ConnectionConfig::Bridge.new if interface_section.bridge_ports && !interface_section.bridge_ports.empty?
+        return ConnectionConfig::Vlan.new if interface_section.etherdevice && !interface_section.etherdevice.empty?
+        return ConnectionConfig::Wireless.new if interface_section.wireless_essid && !interface_section.wireless_essid.empty?
+
+        ConnectionConfig::Ethernet.new # TODO: use type detector to read it from sys
+      end
+
+      def load_generic(config, interface_section)
+        config.bootproto = BootProtocol.from_name(interface_section.bootproto)
+        config.name = interface_section.name || interface_section.device # device is just fallback
+        if config.bootproto == BootProtocol::STATIC
+          # TODO: report if ipaddr missing for static config
+          ipaddr = IPAddress.from_string(interface_section.ipaddr)
+          # Assign first netmask, as prefixlen has precedence so it will overwrite it
+          ipaddr.netmask = interface_section.netmask if interface_section.netmask
+          ipaddr.prefix = interface_section.prefixlen.to_i if interface_section.prefixlen
+          broadcast = interface_section.broadcast && IPAddress.new(interface_section.broadcast)
+          remote = interface_section.remote_ipaddr && IPAddress.new(interface_section.remote_ipaddr)
+          config.ip = IPConfig.new(ipaddr, broadcast: broadcast, remote_address: remote)
+        end
+
+        config.startmode = Startmode.create(interface_section.startmode) if interface_section.startmode
+        config.startmode.priority = interface_section.ifplugd_priority if config.startmode.name == "ifplugd" && interface_section.ifplugd_priority
+        config.mtu = interface_section.mtu.to_i if interface_section.mtu
+        config.ethtool_options = interface_section.ethtool_options if interface_section.ethtool_options
+        config.firewall_zone = interface_section.zone if interface_section.zone
+      end
+
+      def load_wireless(config, interface_section)
+        config.mode = interface_section.wireless_mode if interface_section.wireless_mode
```

Comment:
np: What about having a hash to make the correspondence?

Reply:
I am a bit lost. What you mean here?